### DISCUSSION TITLE
Artifactory - Multiple files upload

### DISCUSSION THREAD

**Discussion by swaroopar** [jreleaser/jreleaser#374] [Q&A] [answered]
Hello, 

We would like to upload multiple files to artifactory using jreleaser. 

Below is the maven configuration -
 
                          <distributions>
                                    <app>
                                        <active>RELEASE</active>
                                        <type>SINGLE_JAR</type>
                                        <artifacts>
                                            <artifact>
                                                <path>target/${project.artifactId}-${project.version}.jar</path>
                                            </artifact>
                                            <artifact>
                                                <path>target/jreleaser/release/CHANGELOG.md</path>
                                            </artifact>
                                        </artifacts>
                                    </app>
                                </distributions>
                              <upload>
                                    <artifactory>
                                        <app>
                                            <active>RELEASE</active>
                                            <uploadUrl>https://artifactory.test/{{projectName}}/{{projectVersion}}/{{artifactfileName}}</uploadUrl>
                                            <authorization>BEARER</authorization>
                                            <files>false</files>
                                            <artifacts>true</artifacts>
                                            <signatures>false</signatures>
                                        </app>
                                    </artifactory>
                                </upload>
                                <environment>
                                    <properties>
                                        <repoOwner>xxxx</repoOwner>
                                        <artifactfileName>${project.artifactId}-${project.version}.jar</artifactfileName>
                                    </properties>
                                </environment>

When the plug is executed, we see the below logs but in Artifactory i see only the jar file. Change log file is not found. 
[INFO] Calculating checksums
[INFO]   [checksum] target/jreleaser/release/CHANGELOG.md.sha256
[INFO]   [checksum] target/test-1.0.2.jar.sha256

[INFO] Signing files
[INFO] Signing is not enabled. Skipping
[INFO] Uploading all artifacts
[INFO]   [artifactory] uploading to app
[INFO]   [artifactory]  - CHANGELOG.md
[INFO]   [artifactory]  - test-1.0.2.jar

Can you please let me know what am I missing with the configuration?

Regards,
Swaroop

**Comment by aalmiray** [comment]
The problem you are facing is the use of the `{{artifactfileName}}` template. The actual name should be `{{artifactFileName}}`, notice the difference in capitalization. If you inspect `target/jreleaser/trace.log` you'll see the full log of the execution, there you may see the resolved URL for each file, in your case both files will point to the same URL and that causes the problem.

**Reply by swaroopar** [reply]
Hi @aalmiray  Thanks for your answer. 
Can I then remove this line in environment/properties? 
`<artifactfileName>${project.artifactId}-${project.version}.jar</artifactfileName>`
Then the "uploadUrl" in artifactory will take all artifacts mentioned in the distributions -> artifacts list?

Also, the "artifactfileName" is mentioned in the docs. Am i mixing up different things in my pom.xml? Please suggest. 

![image](https://user-images.githubusercontent.com/54129659/134275225-70676c0c-ead9-4dc4-b51d-fa7303c0b3cb.png)

**Reply by aalmiray** [reply, marked as answer]
Hi @swaroopar, yes, you can now remove the property definition from environment.
I suspected the docs were wrong, thank you for confirming it. I've just pushed an update correcting the mistake.
